scripts/bulk-mailer/index.js

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const chunk = require('lodash.chunk')
+const config = require('../../config').getProperties()
+const error = require('../../lib/error')
+const P = require('../../lib/promise')
+const readUserRecords = require('./read-user-records')
+const sendEmailBatches = require('./send-email-batches')
+const Senders = require('../../lib/senders')
+const Translator = require('../../lib/senders/translator')
+const UserRecordNormalizer = require('./normalize-user-records')
+const WriteToStreamSenderMock = require('./nodemailer-mocks/stream-output-mock')
+const WriteToDiskSenderMock = require('./nodemailer-mocks/write-to-disk-mock')
+
+
+const bouncesMock = {
+  check: () => P.resolve()
+}
+
+const logMock = {
+  amplitudeEvent: () => {},
+  error: console.error,
+  info: () => {},
+  trace: () => {}
+}
+
+/**
+ * Send an email to users listed in the file `userRecordFilename` using `mailerMethodName`
+ * in batches of `batchSize` with `batchDelayMS` between batches.
```

Comment:
I'm sending 28 messages in stage, with default batchsize (10) and delay (5000ms) and all 28 are sent in 2 to 6 seconds, so it doesn't look like this delay is working properly.

Reply:
Turns out [I'm not making use of this value](https://github.com/mozilla/fxa-auth-server/blob/5c51b92e505b3842515840c47dee9110c3d26bcb/scripts/bulk-mailer/send-email-batches.js#L26). I converted from homegrown processing of the batches to use P.each on each of the batches and forgot to make use of the delay.